subsys/bluetooth/services/bas/bas_bls.c

```diff
@@ -0,0 +1,231 @@
+#define BATTERY_CHARGE_LEVEL_MASK (BIT_MASK(2) << BATTERY_CHARGE_LEVEL_SHIFT)
+#define BATTERY_CHARGE_TYPE_MASK  (BIT_MASK(3) << BATTERY_CHARGE_TYPE_SHIFT)
+#define CHARGING_FAULT_MASK       (BIT_MASK(3) << CHARGING_FAULT_SHIFT)
+
+/*
+ * Bitfield structure: Additional Status
+ *
+ * - Bits 0–1: Service Required
+ * - Bit 2: Battery Fault
+ * - Bits 3–7: Reserved
+ */
+#define SERVICE_REQUIRED_SHIFT 0
+#define BATTERY_FAULT_SHIFT    2
+
+#define SERVICE_REQUIRED_MASK (BIT_MASK(2) << SERVICE_REQUIRED_SHIFT)
+#define BATTERY_FAULT_MASK    (BIT_MASK(1) << BATTERY_FAULT_SHIFT)
+
+void bt_bas_bls_init(void)
+{
+	LOG_INF("Initialise BAS Battery Level Status Module");
+
+	battery_level_status.flags = 0;
+	battery_level_status.power_state = 0;
+
+#if defined(CONFIG_BT_BAS_BLS_IDENTIFIER_PRESENT)
+	/* Set identifier flag */
+	battery_level_status.flags |= BT_BAS_BLS_IDENTIFIER_PRESENT;
+	battery_level_status.identifier = 0;
+#endif
+
+#if defined(CONFIG_BT_BAS_BLS_BATTERY_LEVEL_PRESENT)
+	/* Set battery level flag */
+	battery_level_status.flags |= BT_BAS_BLS_BATTERY_LEVEL_PRESENT;
+	battery_level_status.battery_level = bt_bas_get_battery_level();
+#endif
+
+#if defined(CONFIG_BT_BAS_BLS_ADDITIONAL_STATUS_PRESENT)
+	/* Set additional status flag */
+	battery_level_status.flags |= BT_BAS_BLS_ADDITIONAL_STATUS_PRESENT;
+	battery_level_status.additional_status = 0;
+#endif
+}
+
+/* Convert battery level status to little-endian format */
+static void bt_bas_bls_convert_to_le(struct bt_bas_bls *status)
+{
+	status->power_state = sys_cpu_to_le16(status->power_state);
+
+#if defined(CONFIG_BT_BAS_BLS_IDENTIFIER_PRESENT)
+	status->identifier = sys_cpu_to_le16(status->identifier);
+#endif
+}
+
+static void indicate_cb(struct bt_conn *conn, struct bt_gatt_indicate_params *params, uint8_t err)
+{
+	if (err != 0) {
+		LOG_INF("Indication failed with error %d\n", err);
+	} else {
+		LOG_INF("Indication sent successfully\n");
+	}
+}
+
+static void bt_bas_bls_update_battery_level_status(void)
+{
+	int err;
+	const struct bt_gatt_attr *attr = bt_bas_get_bas_attr(BT_BAS_IDX_BATT_LVL_STATUS_CHAR_VAL);
+
+	if (attr) {
+		struct bt_bas_bls le_battery_level_status = battery_level_status;
+		/* Convert status to little-endian before sending */
+		bt_bas_bls_convert_to_le(&le_battery_level_status);
+		/* Notify/Indicate all connections */
+		ind_params.attr = attr;
+		ind_params.data = &le_battery_level_status;
+		ind_params.len = sizeof(struct bt_bas_bls);
+		ind_params.func = indicate_cb;
+		err = bt_gatt_indicate(NULL, &ind_params);
+		if (err) {
+			LOG_DBG("Failed to send ntf/ind to all connections (err %d)\n", err);
+		}
+	}
+}
+
+ssize_t bt_bas_bls_read_blvl_status(struct bt_conn *conn, const struct bt_gatt_attr *attr,
+				    void *buf, uint16_t len, uint16_t offset)
+{
+	struct bt_bas_bls le_battery_level_status = battery_level_status;
+
+	/* Convert status to little-endian before reading */
+	bt_bas_bls_convert_to_le(&le_battery_level_status);
```

Comment:
You are effectively assigning all 16bit values twice here. 

Why not something simpler (and more explicit) like

```suggestion
	const struct bt_bas_bls bls_le = {
		.flags = battery_level_status.flags,
		.power_state = sys_cpu_to_le16(battery_level_status.power_state),
#if defined(CONFIG_BT_BAS_BLS_IDENTIFIER_PRESENT)
		.identifier = sys_cpu_to_le16(battery_level_status.identifier),
#endif
		.battery_level = battery_level_status.battery_level,
		.additional_status = battery_level_status.additional_status,
	};
```